### DISCUSSION TITLE
Using struct instead of enum for shapes

### DISCUSSION THREAD

**Discussion by TornaxO7** [flxzt/rnote#307] [Ideas]
Hi, I've got a little suggestion here for the [`Shape`](https://github.com/flxzt/rnote/blob/main/rnote-compose/src/shapes/shape.rs#L12) enum.

In my opinion each shape should be a struct and not an enum-entry. Reason: Each
enum has its own functions. If we take a look into the methods of the enum, we
can clearly see, that each enum entry has the "same" functions but with a
different implementation => This can be done with a trait!

What do you think? @flxzt 

## Edit 1
Wait a minute, I just see, that the Shapes are already implemented as a struct. So the enum is just for easier matching?

**Comment by flxzt** [comment]
That pattern is used to store different types in a collection, without using trait objects and dynamic dispatch.

**Reply by TornaxO7** [reply]
hm... ok, I thought that it would be better if we use the trait objects directly but I think it's fine
